apps/s_client.c

```diff
@@ -2218,7 +2219,20 @@ int s_client_main(int argc, char **argv)
                            "Didn't find STARTTLS in server response,"
                            " trying anyway...\n");
             BIO_printf(sbio, "STARTTLS\r\n");
-            BIO_read(sbio, sbuf, BUFSIZZ);
+            mbuf_len = BIO_read(sbio, mbuf, BUFSIZZ);
+            if (mbuf_len < 0) {
+                BIO_printf(bio_err, "BIO_read failed\n");
+                goto end;
+            }
+            mbuf[mbuf_len] = '\0';
+            if (sscanf(mbuf, "%3d ", &numeric) != 1) {
+                BIO_printf(bio_err, "Server does not support STARTTLS.\n");
+                goto shut;
+            }
+            if (numeric != 382) {
```

Comment:
Of course those two bytes can be non-printable, which would be problematic as the can as well be invisible, but so can even longer line. In such case you'd have to save output to file or pipe to `od`... So let's not think about non-printable characters :-)